Source/WebCore/css/CSSSelector.cpp

```diff
@@ -684,6 +685,10 @@ String CSSSelector::selectorText(StringView separator, StringView rightSide) con
             case CSSSelector::PseudoClassType::Scope:
                 builder.append(":scope");
                 break;
+            case CSSSelector::PseudoClassType::State:
+                builder.append(":state(");
+                appendPseudoClassFunctionTail(builder, cs);
+                break;
```

Comment:
```suggestion
            case CSSSelector::PseudoClassType::State:
                builder.append(":state(");
                appendPseudoClassFunctionTail(builder, cs);
                builder.append(')');
                break;
```

Reply:
Would be good to have parsing WPT (See LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing). That WPT could also cover invalid cases like `:state`

Reply:
This doesn't seem resolved, `builder.append(')');` is missing.

Reply:
I don't think this is necessary, because `appendPseudoClassFunctionTail` calls:

```cxx
        builder.append(selector->argument(), ')');
```

Reply:
Ah right, I was mistaken with another part of the code, sorry.